docs/book/src/reference/webhook-for-core-types.md

```diff
@@ -7,47 +7,58 @@ There is an [example](https://github.com/kubernetes-sigs/controller-runtime/tree
 in controller-runtime.
 
 It is suggested to use kubebuilder to initialize a project, and then you can
-follow the steps below to add admission webhooks for core types.
+follow the steps below to add admission webhooks for core types. The example shows
+how to set up a mutating webhook following the controller-runtime's webhook builder.
 
-## Implement Your Handler
+## Implement Your Webhook
 
-You need to have your handler implements the
-[admission.Handler](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission?tab=doc#Handler)
+You need to have your webhook to implement the
+[admission.CustomDefaulter](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission?tab=doc#CustomDefaulter)
```

Comment:
@mythi 

All shows fine less this link. We are not implementing the CustomDefaulter right?
That seems that would be a follow up of this one.

We are here just addressing the breaking changes for the old way to do that.

Reply:
This PR follows changes made in https://github.com/kubernetes-sigs/controller-runtime/commit/927c09e54313d6476b5780b7f029f381d777c149. That flow expects a `CustomDefaulter` is implemented.

Reply:
Hi @mythi 

```
 WithDefaulter(a).
```

You are passing the Resource Structure itself and not the an structure created for the customer. See that we cannot change the docs without change the scaffold to use the CustomDefault and etc

We have a PR to change the scaffold: https://github.com/kubernetes-sigs/kubebuilder/pull/3723 but it has the same issue. See the example of implementation in controller runtime: https://github.com/kubernetes-sigs/controller-runtime/blob/4000e996a202917ad7d40f02ed8a2079a9ce25e9/pkg/builder/webhook.go#L70-L74

So, WDYT about we broke it in phases

1 - Now, just the update the docs to reflect the current scaffold with the previous breaking changes / deprecations addressed 

Then, when we be scaffolding the Custom we can upgrade the docs to use it.

Reply:
> You are passing the Resource Structure itself and not the an structure created for the custom. See that we cannot change the docs without change the scaffold to use the CustomDefault and etc

I'm not sure what this means and that I'd agree with the comment. This doc explicitly says "Given that kubebuilder doesn’t support webhook scaffolding for core types, you have to use the library from controller-runtime to handle it." so I don't see what's the dependency to the rest of the kubebuilder functionality.

This PR merely updates to the controller-runtime example to implement `admission.CustomDefaulter `interface for `podAnnotator` type.

Reply:
We are not using ` admission.CustomDefaulter ` yet in any place but OK. I think you are right we can change this doc and change the scaffolds in another PR.